pkg/gameserverallocations/allocator.go

```diff
@@ -496,6 +496,7 @@ func (c *Allocator) ListenAndAllocate(ctx context.Context, updateWorkerCount int
 	// continued.
 
 	var list []*agonesv1.GameServer
+	var sortKey *uint64 // Pointer so that we can check for nil (and set this value on the first req)
```

Comment:
🤔 any reason the default value of 0 wouldn't suffice (rather than `nil`)? Save you managing pointers?

Reply:
The SortKey() returns 0 if there's an error. An alternative would be to return 1 on error for SortKey(), and then sortKey as a non-pointer uint64, wdyt?

Reply:
That does ask an interesting question - if the sortKey fails (admittedly very unlikely), should we reset sorting? In which case a 0 would be fine. WDYT?

Reply:
Not quite following -- are you saying we should not reset the sortKey when a gameserverallocation generate sortKey fails?

Reply:
I think we're saying the same thing. That the `list` (and everything else) should reset if there was an error generating the `SortKey()`.

In which case, if `var sortKey *uint64` became `var sortKey uint64` - the 0 value would work the same as it does now - in which case, there's no real reason for the pointer.

In the current code, if `SortKey()` errors, it returns 0, which will likely reset the `list` (etc) values (assuming the last `sortKey` > 0).

Did that make more sense?

Reply:
The pointer doesn't work exactly the same as 0 though.

If we got rid of lines 521-523, then the code as written would reset requestCount to 0 on the first request which is incorrect. 

If we changed line 521 to `sortKey == 0` then in the case of an error it would override the existing sortKey to be equal to 0 and continue on with the current batch, and then on the next (passing) request the newSortKey would not be equal to 0 and would reset requestCount and list. This slits the batch into two, with the error being attached to the first batch of requests.

As the code is currently written the error (and any directly adjacent errors) would be in their own batch, and passing batches would be grouped separately.

So maybe we have three options:

1. Totally ignore any errors (besides logging)
2. Split on error and attach the error to the previous batch
3. Split on error where any error(s) is in its own batch

Reply:
> If we got rid of lines 521-523, then the code as written would reset requestCount to 0 on the first request which is incorrect.

Oooh, I see that `requestCount++` on line 506 - goooot it!

Okay yep - that makes sense then, let's leave it as `nil` 👍🏻 thanks for the explanation.

I think as you have it now makes a lot of sense.

Reply:
Actually, would it make more sense to move the `requestCount++` to line 531? As is, requestCount = 0 doesn't make much sense when we're in the middle of processing a request.

Reply:
It's a slight change to the existing logic, but I actually have no strong opinions either way tbh. Your call 😄

Reply:
If you have no strong feelings then I'm for it :)